onnxruntime/core/providers/xnnpack/nn/conv_base.cc

```diff
@@ -326,10 +353,7 @@ bool ConvBase::IsOnnxNodeSupported(const NodeUnit& node_unit, const GraphViewer&
 
     // we only support float and u8 currently
     const auto* x_type = x_arg.TypeAsProto();
-    if (x_type == nullptr ||
-        (x_type->tensor_type().elem_type() != ONNX_NAMESPACE::TensorProto_DataType_FLOAT &&
-         x_type->tensor_type().elem_type() != ONNX_NAMESPACE::TensorProto_DataType_UINT8 &&
-         x_type->tensor_type().elem_type() != ONNX_NAMESPACE::TensorProto_DataType_INT8)) {
+    if (x_type == nullptr || !IsComputeTypeSupported(x_type->tensor_type().elem_type())) {
```

Comment:
Not sure this is generic enough to be in a shared IsComputeTypeSupported function as I thought it was on a per-kernel basis as to whether it supports u8/s8/fp32/fp16.

I could imagine a helper class though that you have a static instance of. Construct with the list of valid types so it's flexible, and it provides a method to check if the tensor element type is in that list.

Reply:
I think using default parameter is simpler to meet it.